### ISSUE TITLE
consumer worker thread not exiting

### ISSUE THREAD

**Issue by FuFikCZ** [Parsely/pykafka#646]
Hello,

I have problem with running code below. Threads aren't stopped after calling SimpleConsumer.stop(). Problem is that fetcher [simpleconsumer.py](https://github.com/Parsely/pykafka/blob/fb9d9191bc2467ecd0b546eba2e24219b35d4d46/pykafka/simpleconsumer.py#L384-L397) is waiting for `fetch()` to finish, but it won't finish due waiting for thread event. SimpleConsumer.stop() sets `self._running = False`, but thread still waits for thread event and won't exit. Solution may be to add call of event.set() in SimpleConsumer.stop().

```python
import threading
import logging
import sys
from pykafka import KafkaClient

log = logging.getLogger('pykafka')
log.setLevel(10)

ch = logging.StreamHandler(sys.stdout)
ch.setLevel(logging.DEBUG)
formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
ch.setFormatter(formatter)

log.addHandler(ch)

client = KafkaClient(zookeeper_hosts='zookeper')
for i in range(10):
    topic = client.topics['topic'.encode('utf-8')]

    consumer = topic.get_simple_consumer(
        auto_commit_enable=False,
        num_consumer_fetchers=1,
        queued_max_messages=500
    )
    consumer.start()

    for j in range(100):
        message = consumer.consume(block=True)
        if message is None:
            break

    consumer.commit_offsets()
    consumer.stop()

print(threading.active_count())
for t in threading.enumerate():
    print(t)
```

**PyKafka version**: 2.5.0
**Kafka version**: 2.11-0.10.1.0

**Comment by emmett9001** [comment]
Thanks for pointing this out @FuFikCZ. It looks at first glance like this might be something that only appears at lower `queued_max_messages` values. You're only reading 100 messages per consumer before killing them - I wonder if a lower `fetch_message_max_bytes` value would fix this issue for you. It's possible that lowering that value to less than the size of 100 messages from your topic could allow some free space in the queue, stopping this deadlock.

All of that said, this still seems very much like a real bug in pykafka that needs to be addressed.

**Comment by FuFikCZ** [comment]
@emmett9001 I have tried lowering `fetch_message_max_bytes`. It won't fix the problem, but number of "zombie" threads is lower. I have a lot of messages which can vary widely in sizes, so I can't rely on it :-( 

It seems that my fix does not solve everything. There may be more deadlocks, because number of threads still rises (but slower).

I will do some more investigation and I will keep you posted.

**Comment by emmett9001** [comment]
I think we might need something like [this](https://github.com/Parsely/pykafka/commit/94532139541448e2aa34e48c530e263a06eeeb59) here. I'll open a PR.

**Comment by emmett9001** [comment]
@FuFikCZ I've merged #683, which should fix this issue. Please reopen or comment if the issue persists.

**Comment by emmett9001** [comment]
Looking back at this change, I'm not sure why I chose to put the `cleanup` call in the autocommitter instead of the fetcher. The example given here doesn't even use autocommitting. #827 updates this PR with what I now think is a better fix, pointed out in #823.
